src/material/autocomplete/autocomplete.spec.ts

```diff
@@ -302,8 +302,12 @@ describe('MatAutocomplete', () => {
       options[1].click();
       fixture.detectChanges();
 
+      const controlValue = fixture.componentInstance.formField._control
+        && fixture.componentInstance.formField._control.value;
+
       expect(fixture.componentInstance.formField.floatLabel)
-          .toEqual('auto', 'Expected label to return to auto state after panel closes.');
+          .toEqual(controlValue != '' && controlValue != null ? 'always' : 'auto',
```

Comment:
Rather than having a ternary here, there should be a separate test case that highlights the issue.